oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/query/MetadataQueryService.java

```diff
@@ -49,8 +47,8 @@ private IMetadataQueryDAO getMetadataQueryDAO() {
         return metadataQueryDAO;
     }
 
-    public List<Service> getAllServices(final String group) throws IOException {
-        return getMetadataQueryDAO().getAllServices(group).stream()
+    public List<Service> listServices(final String layer, final String group) throws IOException {
+        return getMetadataQueryDAO().listServices(layer, group).stream()
```

Comment:
I know you built a `map`-based distinct/merge on DAO level, which should be moved here. And `distinct` is not working anymore as `layer` list added. You should do `reduce` here to merging according to service name.